### PR TITLE
HBW-155 Fix the way ajax request are made in lookup select

### PR DESCRIPTION
**Pull request by mxsmirnoff**

### CODE REVIEW

File: hbw/app/javascript/packs/hbw/components/form/select.js.jsx [hunk 1]
```diff
@@ -200,16 +200,17 @@ modulejs.define('HBWFormSelect',
...
-              label: opt.text,
-              value: opt.id
-            }));
-
-            callback(res);
-          })
+        this.props.env.connection.request({
+          url:    `${url}q=${inputValue}`,
+          method: 'GET'
+        }).then((response) => {
```
Comment (TimAle) [inline hbw/app/javascript/packs/hbw/components/form/select.js.jsx]: Make this chaining style consistent with `.catch`.

File: hbw/app/javascript/packs/hbw/components/form/select.js.jsx [hunk 2]
```diff
@@ -200,16 +200,17 @@ modulejs.define('HBWFormSelect',
...
+          const res = response.map(opt => ({
```
Comment (TimAle) [inline hbw/app/javascript/packs/hbw/components/form/select.js.jsx]: Let's use object destructuring here :)

File: hbw/app/javascript/packs/hbw/components/form/select.js.jsx (hunk 1 above)
Comment (TimAle) [inline hbw/app/javascript/packs/hbw/components/form/select.js.jsx]: It looks fine, but we never used these methods with jQuery.ajax in this project. Did you checked this?

File: hbw/app/javascript/packs/hbw/components/form/select.js.jsx [hunk 3]
```diff
@@ -200,16 +200,17 @@ modulejs.define('HBWFormSelect',
...
-          .then((json) => {
-            const res = json.map(opt => ({
-              label: opt.text,
-              value: opt.id
-            }));
-
-            callback(res);
-          })
+        this.props.env.connection.request({
+          url:    `${url}q=${inputValue}`,
```
Comment (TimAle) [inline hbw/app/javascript/packs/hbw/components/form/select.js.jsx]: You can pass the params in the `data` object I suppose.

File: hbw/app/javascript/packs/hbw/components/form/select.js.jsx (hunk 1 above)
Comment (mxsmirnoff) [reply hbw/app/javascript/packs/hbw/components/form/select.js.jsx]: I checked this, but I can change to `done` for identically

Comment (TimAle) [reply hbw/app/javascript/packs/hbw/components/form/select.js.jsx]: No, it's OK if you checked this :)

File: hbw/app/javascript/packs/hbw/components/form/select.js.jsx [hunk 4]
```diff
@@ -198,23 +198,24 @@ modulejs.define('HBWFormSelect',
...
-
-        fetch(`${url}q=${inputValue}`)
-          .then(response => response.json())
-          .then((json) => {
-            const res = json.map(opt => ({
-              label: opt.text,
-              value: opt.id
-            }));
+        this.props.env.connection.request({
+          url:    `${this.props.params.url}${(this.props.params.url.includes('?') ? '&' : '?')}`,
```
Comment (TimAle) [inline hbw/app/javascript/packs/hbw/components/form/select.js.jsx]: I'm not sure but I think jQuery will handle building url problem completely. Check it please.

**Comment by mxsmirnoff** [comment]
Updated

**Pull request merged**